### ISSUE TITLE
Transformer Language Model Example Wrong Mask

### ISSUE THREAD

**Issue by calclavia** [pytorch/examples#649]
I did a quick test of Transformer decoder in https://github.com/pytorch/examples/blob/master/word_language_model/model.py

It seems like the attention mask is flipped (masking future elements, rather than past).
The correct mask should be:
```
n_ctx = [size of sequence]
mask = torch.triu(torch.ones(n_ctx, n_ctx), 1)
mask = mask.masked_fill(mask == 1, float('-inf'))
```

**Comment by calclavia** [comment]
@zhangguanheng66 Thanks for pointing that out. Is the mask Is the correct mask supposed to be
```
tensor([[0., -inf, -inf],
        [0., 0., -inf],
        [0., 0., 0.]])
```
rather than Pytorch 1.3?
```
tensor([[0., 0., 0.],
        [-inf, 0., 0.],
        [-inf, -inf, 0.]])
```

**Comment by zhangguanheng66** [comment]
see this issue https://github.com/pytorch/pytorch/issues/28508

**Comment by raphael0202** [comment]
If we split the float cast and the masked_fill operations, the Transformer model works as expected:

**Comment by zhangguanheng66** [comment]
@soumith I think we could close this issue as we released v1.3.1. The mask generated by v1.3.1 should be right.

**Comment by stomir** [comment]
I'm not sure if this is a correct place to ask this question, but doesn't this mask mean the network has access to the very word it's supposed to guess?
